exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/ops/InListOps.kt

```diff
@@ -123,3 +121,65 @@ class TripleInListOp<T1, T2, T3>(
         append(")")
     }
 }
+
+/**
+ * Represents an SQL operator that checks if all columns of a `List` [expr] match any of the lists of
+ * values from [list].
+ *
+ * To inverse the operator and check if the `List` of columns is **not** in [list], set [isInList] to `false`.
+ */
+class MultipleInListOp(
+    override val expr: List<Column<*>>,
+    list: Iterable<List<*>>,
+    isInList: Boolean = true
+) : InListOrNotInListBaseOp<List<*>>(expr, list, isInList) {
+    override val columnTypes: List<Column<*>> = expr
+
+    override fun QueryBuilder.registerValues(values: List<*>) {
+        append("(")
+        expr.forEachIndexed { i, expression ->
+            registerArgument(expression.columnType, values[i])
+            if (i != values.lastIndex) append(", ")
+        }
+        append(")")
+    }
+
+    override fun toQueryBuilder(queryBuilder: QueryBuilder) {
+        // SQL Server does not support IN operator with tuples (or any more than 1 expression on the left-hand side)
+        if (currentDialectIfAvailable !is SQLServerDialect) return super.toQueryBuilder(queryBuilder)
+
+        queryBuilder {
+            val iterator = list.iterator()
+            if (!iterator.hasNext()) {
+                if (isInList) {
+                    +FALSE
+                } else {
+                    +TRUE
+                }
+            } else {
+                // Generates compound AND & OR operators for each values list:
+                // WHERE
+                //     ((tester.num_1 = 0) AND (tester.num_2 = 0.0) AND (tester.num_3 = '0') AND (tester.num_4 = 0)) OR
+                //     ((tester.num_1 = 1) AND (tester.num_2 = 1.0) AND (tester.num_3 = '1') AND (tester.num_4 = 1)) OR
+                //     ((tester.num_1 = 2) AND (tester.num_2 = 2.0) AND (tester.num_3 = '2') AND (tester.num_4 = 2))
+
+                // Alternative: EXISTS (SELECT * FROM (VALUES (...), (...), ...) v(...) WHERE v.?=? AND ...)
+                // Built-in exists(AbstractQuery) cannot be used because above row value constructors are not supported
```

Comment:
Interestingly, when using PostgreSQL EXPLAIN, it seems like the standard syntax `WHERE (col_1, col_2, ...) IN ((?,?,...), (?,?,...), ...)` (which isn't supported by SQL Server) ends up with the same execution plan as the implementation below that uses compound `AND` + `OR`:
```bash
Seq Scan on tester  (cost=0.00..49.60 rows=1 width=54)
    Filter: (((num_1 = 0) AND (num_2 = '0'::double precision) AND ((num_3)::text = '0'::text) AND (num_4 = 0)) OR ((num_1 = 1) AND (num_2 = '1'::double precision) AND ((num_3)::text = '1'::text) AND (num_4 = 1)) OR ((num_1 = 2) AND (num_2 = '2'::double precision) AND ((num_3)::text = '2'::text) AND (num_4 = 2)))
```

Whereas the proposed alternative above, using `EXISTS`, has the following plan:
```bash
Hash Semi Join  (cost=0.10..30.52 rows=1 width=54)
    Hash Cond: ((tester.num_1 = "*VALUES*".column1) AND (tester.num_2 = ("*VALUES*".column2)::double precision) AND ((tester.num_3)::text = "*VALUES*".column3) AND (tester.num_4 = "*VALUES*".column4))
    ->  Seq Scan on tester  (cost=0.00..19.90 rows=990 width=54)
    ->  Hash  (cost=0.04..0.04 rows=3 width=72)
         ->  Values Scan on "*VALUES*"  (cost=0.00..0.04 rows=3 width=72)

```